app/src/main/java/at/tugraz/ist/swe/note/NoteStorage.java

```diff
@@ -0,0 +1,93 @@
+package at.tugraz.ist.swe.note;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import at.tugraz.ist.swe.note.database.DatabaseHelper;
+import at.tugraz.ist.swe.note.database.NotFoundException;
+
+public class NoteStorage {
+    public static String DATE_TIME_FORMAT = "yyyy-MM-dd HH:mm:ss";
+    public static SimpleDateFormat DATE_FORMAT = new SimpleDateFormat(DATE_TIME_FORMAT, Locale.getDefault());
+
+    private DatabaseHelper databaseHelper;
+
+    public NoteStorage(DatabaseHelper databaseHelper) {
+        this.databaseHelper = databaseHelper;
+    }
+
+    static private String convertDateToString(Date date) {
+        return DATE_FORMAT.format(date);
+    }
+
+    static private Date convertStringToDate(String date) {
+        try {
+            return DATE_FORMAT.parse(date);
+        } catch(java.text.ParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * @return Returns true if note was inserted, false otherwise.
+     */
+    public boolean save(Note note) {
+        if(note.exists()) {
```

Comment:
Why has the node the exists function? The node doesn't have the information if it exists in the database.
So I would change it to something like that:
`if(isPhantom(note)){...}`

Reply:
The note has the information whether it's exists or not: it is stored in the id
If the id == -1, then it was not saved yet

Reply:
yes, this is clear, but you should do the check anyway on the database because it decides between insert and update.  And it shouldn't trust the node in this case.

Reply:
I have a new solution: why do we need the save and exists functions at all?
Because let's say we are in the NoteCreateActivity, then it is obvious that we want to insert the note -> call NoteStorage.insert()
If we are in the NoteUpdateActivity, then we want to update the note -> call NoteStorage.update()
All other states are not possible.

Reply:
because new/show/edit Note will have the same activity

Reply:
Ok, you're right. But even if it is the same activity, the activity must know whether it was opened to create a new note, or to update an existing note.
This would be easy to detect: If the activity was launched after selecting a note from the note list in MainActivity or if the activity was launched after pressing "create new note" on the MainActivity.
So in my opinion the save method is not necessary.

Reply:
yes, it would be possible.

another approach:
- the node() creates a UUID if it is a new Note.
- the app uses this UUID as unique key.   
- on save the note is always inserted in the database, also if it is an update
- on update all old notes with the same UUID are marked as removed

the database would have all old note saved but marked as deleted, so it would make it possible to have an undo function. 
we should talk about this next week